Alert section for freshness checks (#21676)
dagster-io/dagster: docs/content/concepts/assets/asset-checks/checking-for-data-freshness.mdx

Before continuing, you should be familiar with:

> - [Asset definitions](/concepts/assets/software-defined-assets)

Author: Should this change?

Reviewer: "Asset definitions" is the preferred nomenclature in these kinds of docs now

---

## Alerting on overdue assets (Dagster+ only)

In Dagster+, you can set up alerts to notify you when assets are overdue for an update. By default, freshness checks will fail with a severity of `WARN`, but you can override this to fail with `ERROR`.

To alert on overdue assets, create an alert policy with the following settings:

> <AssetCheckAlerts />

Author: this section looks like it's missing some stuff?

---

## Alert on overdue assets (Dagster+ only)

In Dagster+, you can set up [alerts](/dagster-plus/managing-deployments/alerts) to notify you when an asset is overdue for an update. By default, a freshness check will fail with severity `WARN` if an asset is overdue for an update. Then, in order to alert on overdue assets, you'll want to create an alert policy that triggers on asset check failures with severity `WARN` (or `ERROR` if you've overridden the default severity), for your specific assets which you've defined freshness checks for. Asset alert policies can currently be targeted at specific asset keys, asset groups, or all assets.

To become more familiar with the various ways alert policies can be set up, see the guide on [managing alert policies](/dagster-plus/managing-deployments/alerts/managing-alerts-in-ui).

> <TabGroup>
> <TabItem name="Option 1: Use the alert policy editor UI">
> 1. Navigate to the alert policy editor window in the Dagster UI.

Reviewer 2: ```suggestion
<TabGroup>
<TabItem name="Option 1: Dagster UI">

1. Navigate to the alert policy editor window in the Dagster UI.
```

A few things here:

- We can just say `Dagster UI` here, otherwise it's a mouthful
- A blank line is needed between `TabItem` and the content it contains, or the Markdown won't render correctly
- How do users get to the "alert policy editor window"? I don't think we call it that anywhere else, even in the UI, so this may not make sense to some users

Author: I added some steps to explain how to get to the editor.

Reviewer 2: Did those changes get pushed? Everything looks the same to me atm.

Author: whoops you're right; I never pushed. Did so now.

> 2. Select the `Asset` alert policy type.

Reviewer 2: ```suggestion
2. Select the **Asset** alert policy type.
```

> 3. First select an asset (or set of assets) to target with the alert policy. You can target all assets, a specific asset key, or an asset group.

Reviewer 2: ```suggestion
3. Select an asset (or set of assets) to target with the alert policy. You can target all assets, a specific asset key, or an asset group.
```

4. Check the box for `Failed (WARN)` or `Failed (ERROR)` severity, depending on the severity set for freshness check failure.

> 5. Select a notification service, provide the necessary information, and save the alert policy.

Reviewer 2: ```suggestion
5. Select a [notification service](/dagster-plus/managing-deployments/alerts#supported-notification-services), provide the necessary information, and save the alert policy.
```

</TabItem>
<TabItem name="Option 2: from config">

> Here we provide an example of how you could define an asset-check-targeted alert policy (which would
> also target freshness checks) in your alert policy configuration file:

Reviewer 2: I find this description to be kinda distant from how we describe this approach elsewhere - for example: [https://docs.dagster.io/dagster-plus/managing-deployments/alerts/managing-alerts-cli#how-it-works](https://docs.dagster.io/dagster-plus/managing-deployments/alerts/managing-alerts-cli#how-it-works)

Maybe something like this?

```suggestion
If you prefer to work in code, you can define an asset check-based alert policy in a YAML configuration file and use the `dagster-cloud` CLI to set the policy for the deployment. **Note**: Asset check-based policies also target freshness checks.

[example]

Refer to the [Managing alerts using the dagster-cloud CLI](/dagster-plus/managing-deployments/alerts/managing-alerts-cli#how-it-works) for more information.
```

Author: will do

```yaml
alert_policies:
  - name: "Warning checks on my_asset_key"
    description: "Alert on WARN checks for my_asset_key"
    target:
      asset_key: "my_asset_key"
    severity: "WARN"
    notification_service: "slack"
    notification_config:
      email:
        email_addresses:
          - "foo@bar.com"
          - "bar@foo.com"

> ```

Reviewer 2: Can we replace this with a more realistic example? We get dinged on example quality all the time. It doesn't have to be really complex, just not `foo` `bar`

---

## Alert on overdue assets (Dagster+ only)

In Dagster+, you can set up [alerts](/dagster-plus/managing-deployments/alerts) to notify you when an asset is overdue for an update. By default, a freshness check will fail with severity `WARN` if an asset is overdue for an update. Then, in order to alert on overdue assets, you'll want to create an alert policy that triggers on asset check failures with severity `WARN` (or `ERROR` if you've overridden the default severity), for your specific assets which you've defined freshness checks for. Asset alert policies can currently be targeted at specific asset keys, asset groups, or all assets.

To become more familiar with the various ways alert policies can be set up, see the guide on [managing alert policies](/dagster-plus/managing-deployments/alerts/managing-alerts-in-ui).

<TabGroup>

> <TabItem name="Option 1: Use the alert policy editor UI">

Reviewer 2: Sorry to continue to nitpick here, but can we adopt the approach [for tabs used here](https://docs.dagster.io/dagster-plus/managing-deployments/alerts/email#creating-an-alert-policy), for consistency?

Author: good call

1. Navigate to the **Deployments** in the top navigation
2. Click the **Alerts** tab.
3. Click **+Create alert policy**.
4. Select the **Asset** alert policy type in the **Create alert policy** window.

> 5. Select an asset (or set of assets) to target with the alert policy. You can target all
> 
> assets, a specific asset key, or an asset group. 6. Check the box for **Failed (WARN)** or **Failed (ERROR)** severity, depending on the severity set for freshness check failure. 7. Select a [notification service](/dagster-plus/managing-deployments/alerts#supported-notification-services), provide the necessary information, and save the alert policy. </TabItem> <TabItem name="Option 2: from config">

Reviewer 2: Some wonky formatting here - it should be something like this + running `make mdx-format` to get any nitpicky spacing:

```suggestion
5. Select an asset (or set of assets) to target with the alert policy. You can target all assets, a specific asset key, or an asset group.
6. Check the box for **Failed (WARN)** or **Failed (ERROR)** severity, depending on the severity set for freshness check failure.
7. Select a [notification service](/dagster-plus/managing-deployments/alerts#supported-notification-services), provide the necessary information, and save the alert policy.
</TabItem>
<TabItem name="Option 2: from config">
```

---

## Alert on overdue assets (Dagster+ only)

In Dagster+, you can set up [alerts](/dagster-plus/managing-deployments/alerts) to notify you when an asset is overdue for an update. By default, a freshness check will fail with severity `WARN` if an asset is overdue for an update. Then, in order to alert on overdue assets, you'll want to create an alert policy that triggers on asset check failures with severity `WARN` (or `ERROR` if you've overridden the default severity), for your specific assets which you've defined freshness checks for. Asset alert policies can currently be targeted at specific asset keys, asset groups, or all assets.

To become more familiar with the various ways alert policies can be set up, see the guide on [managing alert policies](/dagster-plus/managing-deployments/alerts/managing-alerts-in-ui).

<TabGroup>
<TabItem name="In the UI">

> 1. Navigate to the **Deployments** in the top navigation

Reviewer 2: ```suggestion
1. Click the **Deployments** in the top navigation.
```

2. Click the **Alerts** tab.
3. Click **+Create alert policy**.
4. Select the **Asset** alert policy type in the **Create alert policy** window.

> 5. Select an asset (or set of assets) to target with the alert policy. You can target all
> 
> assets, a specific asset key, or an asset group. 6. Check the box for **Failed (WARN)** or **Failed (ERROR)** severity, depending on the severity set for freshness check failure. 7. Select a [notification service](/dagster-plus/managing-deployments/alerts#supported-notification-services), provide the necessary information, and save the alert policy. </TabItem> <TabItem name="Via config">

Reviewer 2: ```suggestion
5. Select an asset (or set of assets) to target with the alert policy. You can target all assets, a specific asset key, or an asset group. 
6. Check the box for **Failed (WARN)** or **Failed (ERROR)** severity, depending on the severity set for freshness check failure. 
7. Select a [notification service](/dagster-plus/managing-deployments/alerts#supported-notification-services) and provide the necessary information.
8. When finished, click **Save**.

</TabItem>
<TabItem name="Via config">
```

Some weird formatting here
